src/Record.js

```diff
@@ -28,11 +28,34 @@ import { asImmutable } from './methods/asImmutable';
 
 import invariant from './utils/invariant';
 import quoteString from './utils/quoteString';
+import { isImmutable } from './predicates/isImmutable';
+
+function convertDefaultValues(defaultValues) {
+  if (isRecord(defaultValues)) {
+    return defaultValues.toObject();
+  }
+
+  if (isImmutable(defaultValues)) {
+    throw new Error(
+      'Can not call `Record` an immutable Collection as default values. You may want to use a plain javascript object instead.'
```

Comment:
A "with" is missing now. And I still think "you may want" is a bit silly. We know what the user needs to do. We are telling them.